### PR TITLE
Add initial document for generated schemas

### PR DESCRIPTION
**Pull request by acmiyaguchi**
This addresses #398 by describing generated schemas in the data pipeline.

### CODE REVIEW

**Comment by acmiyaguchi** [comment]
@jklukas r? There are still a few sections to fill in, but I'm wondering if you have any feedback.

File: src/concepts/pipeline/schemas.md [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+end
+```
+
+**Figure**: _Template schemas are built locally to generate static JSON Schema. On a regular basis,
+the Mozilla Schema Generator is run to generate BigQuery schemas._
+
+Refer to [Sending a Custom Ping](../../cookbooks/new_pings.md) for an in-depth guide for adding new
+schemas to the repository.
+
+### Mozilla Schema Generator
```
Comment (fbertsch) [inline src/concepts/pipeline/schemas.md]: A few key points here:
- JSON schemas are generated and used by the transpiler for BQ schema generation, but _not_ for validation
- Schema aliases are generated here
- A few fields in the environment are altered as well

File: src/concepts/pipeline/schemas.md [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+
+### Airflow
+
+## Data Ingestion
+
+### Validation
+
+Data that fails validation is sent to the `payload_bytes_error` table. Each row contains
```
[the remark this answers is not in the document]
Comment (acmiyaguchi) [reply src/concepts/pipeline/schemas.md]: The monitoring views are inaccessible at the moment, too, if I'm recalling the authorized view situation correctly.

File: src/concepts/pipeline/schemas.md [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+
+```sql
+SELECT
+  payload.processes.parent.keyed_scalars.a11y_theme
+FROM `moz-fx-data-shared-prod`.telemetry.main
+WHERE date(submission_timestamp) = date_sub(current_date, interval 1 day)
+LIMIT 1
+```
+
+#### Data Normalization
```
[the remark this answers is not in the document]
Comment (acmiyaguchi) [reply src/concepts/pipeline/schemas.md]: We are altering the data by introducing pseudo-maps, anonymous tuples, and their ilk. But I see what you mean, we're modifying the structure (and types) of the data to match the schema, as opposed to modifying their contents.

File: src/concepts/pipeline/schemas.md [hunk 4]
```diff
@@ -0,0 +1,307 @@
...
+
+## Overview
+
+Schemas describe the structure of ingested data. They are used in the pipeline to validate the types
+and values of data, or to define a table schema in a data store. We use a repository of JSON Schemas
+to sort incoming data into [`decoded` and `error`
+datasets](../../cookbooks/bigquery.md#projects-with-bigquery-datasets). We also generate BigQuery
+table schemas nightly from the JSON Schemas.
+
+This section is intended for those who want modify the process of generating and applying schemas
```
Comment (jklukas) [inline src/concepts/pipeline/schemas.md]:
```suggestion
This section is intended for those who want to modify the process of generating and applying schemas
```

File: src/concepts/pipeline/schemas.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (jklukas) [reply src/concepts/pipeline/schemas.md]: Per discussion last week, we will be removing the `ip` field and making the error table user-accessible.

File: src/concepts/pipeline/schemas.md [hunk 5]
```diff
@@ -0,0 +1,307 @@
...
+`error_type` | `ParsePayload` | The `ParsePayload` type is associated with schema validation or corrupt data
+`exception_class` | `org.everit.json.schema.ValidationException `| Java [JSON Schema Validator](https://github.com/everit-org/json-schema) library
+`job_name` | `decoder-0-0121192636-9c56ac6a` | Name of the Dataflow job that can be used to determine the version of the schema artifact
+
+### Decoding
+
+The BigQuery schemas are used to normalize relevant payload data and determine additional
+properties. Normalization involves renaming field names and transforming certain types of data.
+Snake casing is employed across all schemas, which ensures that nested structures can always be
+accessed using the dot operator. Some data must be transformed before insertion, such as map-types
```
Comment (jklukas) [inline src/concepts/pipeline/schemas.md]: I don't see how snake casing relates to being able to use the dot operator for field access. Our snake casing implementation does also handle transforming special characters to underscore, but perhaps that should be called out separately.

File: src/concepts/pipeline/schemas.md [hunk 6]
```diff
@@ -0,0 +1,307 @@
...
+(a.k.a. dictionaries in Python), due to limitations in BigQuery data representation. Other data may
+not be specified in the schema, and instead placed into a specially constructed column named
+`additional_properties`.
+
+#### Column Name Normalization
+
+The reference [Python
+implementation](https://github.com/acmiyaguchi/test-casing/blob/master/src/main.py) of the snake
+casing algorithm been implemented in the decoder and transpiler using [a shared
```
Comment (jklukas) [inline src/concepts/pipeline/schemas.md]:
```suggestion
casing algorithm is ensured to be compatible with the implementations in the decoder and transpiler using [a shared
```

File: src/concepts/pipeline/schemas.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (acmiyaguchi) [reply src/concepts/pipeline/schemas.md]: Monitoring views can be added later, if they make sense in this section.

File: src/concepts/pipeline/schemas.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (acmiyaguchi) [reply src/concepts/pipeline/schemas.md]: I chose `Data Structure Normalization`, since this is a structural transformation.

**Pull request merged**